### PR TITLE
Misc. MIDI fixes and refactoring

### PR DESCRIPTION
**Pull request by ceski-1**
This is a follow-up to https://github.com/fabiangreffrath/woof/pull/1635.

Changes:

- Fix missing second parameter for some types of short messages (fixes broken macOS music with real SC-55, likely other devices too).
- Move midi file loading/parsing from main thread to midi thread.
- Include SysEx status byte when parsing midi file instead of allocating during midi sequencing.
- Check SysEx event types once during midi file parsing instead of during midi sequencing.
- ~~Remove OPL3 meta message errors.~~ Reverted.
- Rename reset type "None" to "No SysEx".
- Refactoring and clean up.

Also:

- Add "auto" delay after reset when `midi_reset_delay` is `-1`. By default, it works out to 200 ms but will vary depending on the `midi_*` settings. This mitigates stuttering and wrong instruments at the start of a song with a real SC-55, Nuked SC-55, and various devices when using Linux, especially through Wine. Experimental, I may revert this later.

### CODE REVIEW

File: src/i_oplmusic.c [hunk 1]
```diff
@@ -1245,24 +1245,11 @@ static void MetaSetTempo(unsigned int tempo)
...
-        case MIDI_META_TRACK_NAME:
-        case MIDI_META_INSTR_NAME:
-        case MIDI_META_LYRICS:
-        case MIDI_META_MARKER:
-        case MIDI_META_CUE_POINT:
-        case MIDI_META_SEQUENCER_SPECIFIC:
-            break;
-
-        case MIDI_META_SET_TEMPO:
+        case MIDI_SET_TEMPO:
```
Comment (rfomin) [inline src/i_oplmusic.c:1252]: I like all the changes in this PR, impressive work! But with so many cosmetic changes, are we giving up sync with Chocolate Doom?

Comment (ceski-1) [reply src/i_oplmusic.c:1252]: That's a good point. I may have gone overboard with the cosmetic changes, honestly. Maybe I should revert some and update Chocolate once things stabilize a bit more (so we don't bug turol too much).

**Comment by liPillON** [comment]
I tried the new midi subsystem using the latest builds (win64)
Since this is still being worked on, I thought I'd share a minor complaint.. 
I'm using the comments for this PR instead of the other one since that has been merged, sorry...

Anyway, I've noticed that:
- the `midi_player` variable is no longer used to specify which backend should be used
- the `soundfont_path` variable is no longer used to specify which available SF2 file Fluidsynth shoud load
- the `midi_player_menu` seems to be working the same as before, corresponding to the chosen backend/device combo out of all those available in the system

This prevents users from creating portable Woof setups, since different machines would probably have different available midi devices: the Fluidsynth+TIMGM6MB combo might correspond to '1' in one system and to N in another system which happens to have X midi devices and Y soundfonts installed.

In the current release, by setting `midi_player 1` and `soundfont_path "path\to\file.sf2"` one can be reasonably certain of the results: Woof will use the Fluidsynth backend and the specified soundfont no matter how many enumerated midi players a system could have.

**Comment by rfomin** [comment]
> In the current release, by setting `midi_player 1` and `soundfont_path "path\to\file.sf2"` one can be reasonably certain of the results: Woof will use the Fluidsynth backend and the specified soundfont no matter how many enumerated midi players a system could have.

Maybe, if all systems use the same OS. For example, the release version in Linux would have a different soundfont path and fewer midi players available. So I decided to simplify it.

I'm surprised anyone noticed. I'll think about a better config system.

**Comment by liPillON** [comment]
> Maybe, if all systems use the same OS.
yeah I forgot to specify "on windows systems"

> I'm surprised anyone noticed.
I might very well be the only one ;)
It occured to me since I use to test new builds on my main system and then (after having tweaked the config) try them on my other pcs and this time I was getting a different midi playback experience on each.

> I'll think about a better config system.
Thank you!

File: src/i_oplmusic.c (hunk 1 above)
Comment (ceski-1) [reply src/i_oplmusic.c]: Okay, I dropped most of the less important cosmetic changes.

Comment (rfomin) [reply src/i_oplmusic.c]: Looks good, thank you.

File: src/i_midimusic.c [hunk 2]
```diff
@@ -280,16 +317,30 @@ static void ResetPitchBendSensitivity(void)
...
+// devices (e.g. SC-55).
+
+static void ResetDelayBytes(uint32_t length)
+{
+    if (midi_reset_delay == -1)
+    {
+        // MIDI transfer period is 320 us per byte (MIDI 1.0 Electrical Spec
+        // Update CA-033, page 2). Round up value.
+        const uint32_t delay = length * 320 / 1000 + (length * 320 % 1000 != 0);
+        SDL_Delay(delay);
```
Comment (rfomin) [inline src/i_midimusic.c:343]: How about this?
```diff
diff --git a/src/i_midimusic.c b/src/i_midimusic.c
index 40badcc..8c4c8c2 100644
--- a/src/i_midimusic.c
+++ b/src/i_midimusic.c
@@ -376,8 +376,7 @@ static void ResetDelayBytes(uint32_t length)
     {
         // MIDI transfer period is 320 us per byte (MIDI 1.0 Electrical Spec
         // Update CA-033, page 2). Round up value.
-        const uint32_t delay = length * 320 / 1000 + (length * 320 % 1000 != 0);
-        SDL_Delay(delay);
+        I_SleepUS(320 * length);
     }
 }
 
```
I like the auto delay option, it seems to help with the "popping" problem.

Comment (ceski-1) [reply src/i_midimusic.c]: > How about this?

Oh yeah, I forgot about that function, thanks. Fixed in https://github.com/fabiangreffrath/woof/pull/1657/commits/223ebd0b5e492cf4c8f39e6bd47e65ddb60d15a8.

>I like the auto delay option, it seems to help with the "popping" problem.

It seems to work well so far.

**Review by rfomin — APPROVED** [review]
Works great, thanks!

**Pull request merged**